storage/src/vespa/storage/persistence/filestorage/filestorhandlerimpl.h

```diff
@@ -109,47 +110,47 @@ class FileStorHandlerImpl : private framework::MetricUpdateHook,
+            std::lock_guard guard(*_lock);
             return _queue->size();
         }
         void release(const document::Bucket & bucket, api::LockingRequirements reqOfReleasedLock,
                      api::StorageMessage::Id lockMsgId);
 
         // Subsumes isLocked
-        bool operationIsInhibited(const vespalib::MonitorGuard&, const document::Bucket&,
+        bool operationIsInhibited(const monitor_guard &, const document::Bucket&,
                                   const api::StorageMessage&) const noexcept;
-        bool isLocked(const vespalib::MonitorGuard &, const document::Bucket&,
+        bool isLocked(const monitor_guard &, const document::Bucket&,
                       api::LockingRequirements lockReq) const noexcept;
 
-        void lock(const vespalib::MonitorGuard &, const document::Bucket & bucket,
+        void lock(const monitor_guard &, const document::Bucket & bucket,
                   api::LockingRequirements lockReq, const LockEntry & lockEntry);
 
         std::shared_ptr<FileStorHandler::BucketLockInterface> lock(const document::Bucket & bucket, api::LockingRequirements lockReq);
         void failOperations(const document::Bucket & bucket, const api::ReturnCode & code);
 
-        FileStorHandler::LockedMessage getNextMessage(uint32_t timeout, Disk & disk);
+        FileStorHandler::LockedMessage getNextMessage(vespalib::duration timeout, Disk & disk);
         void dumpQueue(std::ostream & os) const;
         void dumpActiveHtml(std::ostream & os) const;
         void dumpQueueHtml(std::ostream & os) const;
-        vespalib::Monitor & exposeLock() { return _lock; }
+        std::mutex & exposeLock() { return *_lock; }
         PriorityQueue & exposeQueue() { return *_queue; }
         BucketIdx & exposeBucketIdx() { return bmi::get<2>(*_queue); }
         void setMetrics(FileStorStripeMetrics * metrics) { _metrics = metrics; }
     private:
-        bool hasActive(vespalib::MonitorGuard & monitor, const AbortBucketOperationsCommand& cmd) const;
+        bool hasActive(monitor_guard & monitor, const AbortBucketOperationsCommand& cmd) const;
         // Precondition: the bucket used by `iter`s operation is not locked in a way that conflicts
         // with its locking requirements.
-        FileStorHandler::LockedMessage getMessage(vespalib::MonitorGuard & guard, PriorityIdx & idx,
+        FileStorHandler::LockedMessage getMessage(monitor_guard & guard, PriorityIdx & idx,
                                                   PriorityIdx::iterator iter);
         using LockedBuckets = vespalib::hash_map<document::Bucket, MultiLockEntry, document::Bucket::hash>;
         const FileStorHandlerImpl      &_owner;
         MessageSender                  &_messageSender;
         FileStorStripeMetrics          *_metrics;
-        vespalib::Monitor               _lock;
+        std::unique_ptr<std::mutex>                _lock;
+        std::unique_ptr<std::condition_variable>   _cond;
```

Comment:
Out of curiosity, why the added indirection?

Reply:
Because the stripe need to be movable. Might be better to avoid that though. Will do next round.